muspinsim/input/keyword.py

```diff
@@ -174,15 +174,20 @@ def __init__(self, block=[], args=[], variables=[]):
     def _store_values(self, block):
         self._values = []
         for v in block:
-            b = [
-                [
-                    LarkExpression(
-                        tk, variables=self._variables, functions=self._functions
-                    )
-                    for tk in lark_tokenize(l)
+            try:
+                b = [
+                    [
+                        LarkExpression(
+                            tk, variables=self._variables, functions=self._functions
+                        )
+                        for tk in lark_tokenize(l)
+                    ]
+                    for l in v
                 ]
-                for l in v
-            ]
+            except LarkExpressionError as e:
+                raise ValueError(
```

Comment:
Getting the line number of input is non-trivial because we use regex to parse the file into blocks before validating - stripping out whitespaces and comments. Thus we can't accurately say which line the error will be on. 

I could add a variable to keep track of the line number where we find the start of a new block. Producing an error like:

``` Error occurred when parsing keyword hyperfine (block starting at line 10):
Invalid variable: 'MHz', valid functions are ['deg', 'e', 'inf', 'pi']
 ```

Reply:
The block line number is fine, that output is _way_ easier to read now